File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/csharp/Handlers.cs
public class Handlers {
    private Service service = new Service();

    public Dog HandleCreateDog(Request request) {
        string name = request.Get("name");
        return service.CreateDog(name);
    }

    public Cat HandleCreateCat(Request request) {
        string name = request.Get("name");
        return service.CreateCat(name);
    }

    public int HandleTransfer(Request request) {
        Shelter shelter = new Shelter();
        Dog dog = new Dog(request.Get("name"));
        service.TransferAnimal(dog, shelter);
        return shelter.Count();
    }

    public object HandleList(Request request) {
        return service.ListAnimals();
    }

    public bool Validate(Request request) {
        if (request.Get("name") == null) {
            throw new ArgumentException("name required");
        }
        return true;
    }
}

File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/java/Handlers.java
public class Handlers {
    private Service service = new Service();

    public Dog handleCreateDog(Request request) {
        String name = request.get("name");
        return service.createDog(name);
    }

    public Cat handleCreateCat(Request request) {
        String name = request.get("name");
        return service.createCat(name);
    }

    public int handleTransfer(Request request) {
        Shelter shelter = new Shelter();
        Dog dog = new Dog(request.get("name"));
        service.transferAnimal(dog, shelter);
        return shelter.count();
    }

    public Object handleList(Request request) {
        return service.listAnimals();
    }

    public boolean validate(Request request) {
        if (request.get("name") == null) {
            throw new IllegalArgumentException("name required");
        }
        return true;
    }
}
